components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt

```diff
@@ -0,0 +1,122 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.awesomebar.provider
+
+import mozilla.components.browser.search.SearchEngine
+import mozilla.components.browser.search.suggestions.SearchSuggestionClient
+import mozilla.components.concept.awesomebar.AwesomeBar
+import mozilla.components.feature.search.SearchUseCases
+import mozilla.components.support.base.log.logger.Logger
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.URL
+
+/**
+ * A [AwesomeBar.SuggestionProvider] implementation that provides a suggestion containing search engine suggestions (as
+ * chips) from the passed in [SearchEngine].
+ */
+class SearchSuggestionProvider(
+    private val searchEngine: SearchEngine,
+    private val searchUseCase: SearchUseCases.DefaultSearchUrlUseCase
+) : AwesomeBar.SuggestionProvider {
+    private val client = if (searchEngine.canProvideSearchSuggestions) {
+        SearchSuggestionClient(searchEngine, { url -> fetch(url) })
```

Comment:
nit: As the lambda is the last parameter  you could pull it the  outside of the parenthesis.
```kotlin
SearchSuggestionClient(searchEngine){ url -> 
    fetch(url) 
} 
```

Reply:
Oh yeah, that makes sense. I'll update the patch.